firebase-installations/customer-lint-checks/src/main/java/com/google/firebase/installations/lint/IncompatibleIidVersionDetector.kt

```diff
@@ -0,0 +1,102 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.installations.lint
+
+import com.android.tools.lint.detector.api.Detector
+
+import com.android.builder.model.MavenCoordinates
+import com.android.tools.lint.detector.api.Category
+import com.android.tools.lint.detector.api.Context
+import com.android.tools.lint.detector.api.Implementation
+import com.android.tools.lint.detector.api.Issue
+import com.android.tools.lint.detector.api.Location
+import com.android.tools.lint.detector.api.Scope
+import com.android.tools.lint.detector.api.Severity
+
+class IncompatibleIidVersionDetector : Detector() {
+    companion object Issues {
+        private val IMPLEMENTATION = Implementation(
+                IncompatibleIidVersionDetector::class.java,
+                Scope.EMPTY
+        )
+
+        @JvmField
+        val INCOMPATIBLE_IID_VERSION = Issue.create(
+                id = "IncompatibleIidVersion",
+                briefDescription = "FirebaseInstanceId version is not compatible with FirebaseInstallations",
+
+                explanation = """
+           These versions of FirebaseInstanceId and FirebaseInstallations can not be used in parallel.
+           Please update your Firebase Instance ID to a version 20.1.0 or later.
```

Comment:
There seems to be an inconsistent Inconsistent use of versions!
Sometimes it's 20.1.0, sometimes 20.1.1 (see below).
I assume that you want to prevent usage of <20.1.0 and recommend usage of v20.1.1...

I think, you would make your life easier by just marking everything before v20.1.1 as incompatible and keep it consistent.

Reply:
Agreed. Thanks for catching this.

Reply:
This explanation is not the part that developers are most likely to see, instead this is the error they will see in their build:

```
/Users/vkryachko/Code/firebase/firebase-android-sdk/firebase-inappmessaging-display: Error: Incompatible IID version found in variant debug: com.google.firebase:firebase-iid:20.0.1@aar [IncompatibleIidVersion]
```

With that in mind I think it could be improved in a few ways:
1. Most likely developers don't have a direct dependency on iid and don't even know what iid is, it might make sense mention that the dependency may be transitive and how it most likely ended up in their build
2. "@aar" part may confuse developers, consider dropping it
3. Consider adding a call for action here like how exactly the problem can be solved with a code example to copy paste into their build file.